Remotion/ObjectBinding/Web.ClientScript/Scripts/BocAutoCompleteReferenceValue.ts

```diff
@@ -0,0 +1,343 @@
+        {
+          this._textbox.attr('data-label-id-index', combobox.attr('data-label-id-index'));
+          combobox.removeAttr('data-label-id-index');
+        }
+      }
+      else
+      {
+        this._textbox.attr('aria-labelledby', internetExplorerScreenReaderLabelID);
+      }
+
+      if (combobox.attr('aria-describedby') !== undefined)
+      {
+        this._textbox.attr('aria-describedby', combobox.attr('aria-describedby'));
+        combobox.removeAttr('aria-describedby');
+      }
+
+      combobox = this._textbox;
+    }
+
+    const errorHandler = (error: Sys.Net.WebServiceError) =>
+    {
+      this.SetError(resources.LoadIconFailedErrorMessage);
+    };
+
+    this._textbox.autocomplete(this._controlServiceUrl, 'Search', 'SearchExact',
+      {
+        extraParams: searchContext,
+        isAutoPostBackEnabled: this._isAutoPostBackEnabled,
+        nullValue: this._nullValueString, // the hidden field value indicating that no value has been selected
+        searchStringValidationParams:
+        {
+          inputRegex: new RegExp(searchStringValidationInfo.ValidSearchStringRegex),
+          dropDownTriggerRegex: new RegExp(searchStringValidationInfo.ValidSearchStringForDropDownRegex),
+          dropDownTriggerRegexFailedMessage: searchStringValidationInfo.SearchStringForDropDownDoesNotMatchRegexMessage,
+          getDropDownSearchString: (searchString: string) =>
+          {
+            return searchStringValidationInfo.IgnoreSearchStringForDropDownUponValidInput ? this.GetDropDownSearchStringForValidInput(searchString) : searchString;
+          }
+        },
+        max: completionSetCount, // Set query limit
+
+        dropDownDisplayDelay: dropDownDisplayDelay,
+        dropDownRefreshDelay: dropDownRefreshDelay,
+        selectionUpdateDelay: selectionUpdateDelay,
+
+        noDataFoundMessage: resources.NoDataFoundMessage,
+        autoFill: true,
+        matchContains: true,
+        combobox: combobox,
+        selectListID: this._selectListID,
+        informationPopUpID: this._informationPopUpID,
+        dropDownButtonID: button.attr('id'),
+        inputAreaClass: 'content',
+        // this can be set to true/removed once the problem is fixed that an empty this.textbox still selects the first element, making it impossible to clear the selection
+        selectFirst: function (inputValue: string, searchTerm: string)
+        {
+          return inputValue.length > 0;
+        },
+        dataType: 'json',
+        parse: function (data: BocAutoCompleteReferenceValue_Item[])
+        {
+          return $.map(data, function (row)
+          {
+            row.IsAnnotation = row.UniqueIdentifier === nullValueString;
+
+            return {
+              data: row,
+              value: row.UniqueIdentifier,
+              result: row.IsAnnotation ? '' : row.DisplayName
+            };
+          });
+        },
+        formatItem: function (item: JQuery) //What we display on input box
```

Comment:
This review comment is still unresolved:
> I think this should be BocAutoCompleteReferenceValue_Item, not JQuery

Also, interesting question: why does this compile? does the JQuery type have `any` quality?

Reply:
Yes, I overlooked this comment, thanks! I believe the JQuery type has some type of "catch-all" member